File: TCS_DIGITAL_1.py
'''
Write a program to find the count of numbers which consists of unique digits.
Input:
Input consist of two Integer lower and upper value of an range
Output:
Output consists of single line, print the count of unique digits in given range. Else
Print"No Unique Number"
'''
def uniqueDigits(num):
    num = str(num)// 23456
    x = set(num)// 23456
    return len(num)==len(x)
t = int(input())
lis =[0]
# lis[i] = lis[i-1] 
for i in range(1,100001):
    if(uniqueDigits(i)):
        lis.append(lis[-1]+1)
    else:
        lis.append(lis[-1])
for i in range(t):
    start,end = map(int,input().split(" "))
    print(lis[end]-lis[start-1])

    

File: TCS_DIGITAL_1.c
/*
Write a program to find the count of numbers which consists of unique digits.
Input:
Input consist of two Integer lower and upper value of an range
Output:
Output consists of single line, print the count of unique digits in given range. Else
Print"No Unique Number"

*/
#include<stdio.h>
int uniqueDigitsHash(int num)// 6
{// 23432%10 = 2
    int hash[10]={0};
    while(num)
    {
        if(hash[num%10]==1)
            return 0;
        hash[num%10]++;
        num/=10;
    }
    return 1;
}
int uniqueDigitsBit(int num)
{
    short int x=0,y;
    while(num)
    {
        int k = num%10;
        y = 1 << k;
        if((x&y)==0)
            x = x|y;
        else
            return 0;
        num/=10;
    }
    return 1;
}   
int main()
{
    int t,start,end;
    scanf("%d",&t);
    int arr[100001];
    arr[0]=1;
    for(int i=1;i<100001;i++) // 10^5*6 < 10^6   10^5 * 10 = 10^6
    {
        if(uniqueDigitsBit(i)==1)
        {
            arr[i] = arr[i-1]+1;
        }
        else
        {
            arr[i] = arr[i-1];
        }
        
    }
    for(int i=0;i<t;i++)
    {
        scanf("%d %d",&start,&end);
        printf("%d\n",arr[end]-arr[start-1]);
    }
}